src/memefactory/server/syncer.cljs

```diff
@@ -109,178 +111,180 @@
+                                       :meme/comment comment
+                                       :meme/image-hash image-hash
+                                       :meme/title title}))
+         (schedule-meme-number-assigner registry-entry (inc (- (bn/number challenge-period-end)
+                                                               (server-utils/now-in-seconds))))
+         (when search-tags
+           (doseq [t (into #{} search-tags)]
+             (db/tag-meme! {:reg-entry/address registry-entry :tag/name t}))))))))
 
 (defn param-change-constructed-event [_ {:keys [:args]}]
-  (try-catch
+  (safe-go
    (let [{:keys [:registry-entry :creator :version :deposit :challenge-period-end :db :key :value :timestamp :meta-hash]} args
-         hash (web3/to-ascii meta-hash)]
-      (add-registry-entry {:reg-entry/address registry-entry
-                           :reg-entry/creator creator
-                           :reg-entry/version version
-                           :reg-entry/created-on timestamp
-                           :reg-entry/deposit (bn/number deposit)
-                           :reg-entry/challenge-period-end (bn/number challenge-period-end)}
-                          timestamp)
-      (promise-> (server-utils/get-ipfs-meta @ipfs/ipfs hash)
-                 (fn [param-meta]
-                   (let [{:keys [reason]} param-meta]
-                     (db/insert-or-replace-param-change!
-                      {:reg-entry/address registry-entry
-                       :param-change/db db
-                       :param-change/key key
-                       :param-change/value (bn/number value)
-                       :param-change/original-value (:param/value (first (db/get-params db [key])))
-                       :param-change/reason reason
-                       :param-change/meta-hash hash})))))))
+         hash (web3/to-ascii meta-hash)
+         {:keys [reason]} (server-utils/get-ipfs-meta @ipfs/ipfs hash)]
+     (add-registry-entry {:reg-entry/address registry-entry
+                          :reg-entry/creator creator
+                          :reg-entry/version version
+                          :reg-entry/created-on timestamp
+                          :reg-entry/deposit (bn/number deposit)
+                          :reg-entry/challenge-period-end (bn/number challenge-period-end)}
+                         timestamp)
+     (db/insert-or-replace-param-change!
+      {:reg-entry/address registry-entry
+       :param-change/db db
+       :param-change/key key
+       :param-change/value (bn/number value)
+       :param-change/original-value (:param/value (first (db/get-params db [key])))
+       :param-change/reason reason
+       :param-change/meta-hash hash}))))
 
 (defn param-change-applied-event [_ {:keys [:args]}]
   (let [{:keys [:registry-entry :timestamp]} args]
-    (js/Promise.resolve (db/update-param-change! {:reg-entry/address registry-entry
```

Comment:
shouldn't it still be wrapped in the macro?

Reply:
This doesn't need it since it doesn't have async call in its body